include/ginkgo/core/base/range.hpp

```diff
@@ -132,6 +132,13 @@ GKO_ATTRIBUTES GKO_INLINE constexpr bool operator!=(const span& first,
     return !(first == second);
 }
 
+/**
+ * A span that is used exclusively for local numbering.
+ */
+struct local_span : span {
+    using span::span;
+};
```

Comment:
```suggestion
using local_span = span;
```

Reply:
That doesn't work, since then both types will be the same. I actually need `local_span` to be different from `span`, because this forces users to explicitly use `local_span` instead of `span`.

Reply:
Yes. I think this should make it clearer to the users than using `span local_rows`.